### PR TITLE
drop virtualenv test dependency, improve README

### PR DESCRIPTION
**Pull request by Bastian-Krause**
Small collection of testing related improvements:
- github/README: drop python virtualenv test dependency
- README: clarify which calls must be run in the venv
- README: mention test dependencies

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -41,7 +41,7 @@ jobs:
 
     - name: Install test dependencies
       run: |
-        sudo apt-get install libgirepository1.0-dev python3-virtualenv nginx-full
```
Comment (ejoerns) [inline .github/workflows/tests.yml:44]: Was `python3-virtualenv` ever used acutally?

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -60,7 +60,7 @@ jobs:
     - name: Run test suite
       run: |
         ./test/wait-for-hawkbit-online
-        ASAN_OPTIONS=fast_unwind_on_malloc=0 dbus-run-session -- pytest -v
+        ASAN_OPTIONS=fast_unwind_on_malloc=0 dbus-run-session -- python -m pytest -v
```
Comment (ejoerns) [inline .github/workflows/tests.yml:63]: How relates this to venv usage?

File: .github/workflows/tests.yml (hunk 1 above)
Comment (Bastian-Krause) [reply .github/workflows/tests.yml:44]: It was described in the README.rst. It is not used in the workflow.

File: .github/workflows/tests.yml (hunk 2 above)
Comment (Bastian-Krause) [reply .github/workflows/tests.yml:63]: If anyone has an idea *why* that is, I'm also happy to add that.

Comment (Bastian-Krause) [reply .github/workflows/tests.yml:63]: > With `virtualenv` a `venv/bin/pytest` exists, but not with `venv`. I should have probably added that to the commit message.

This is wrong. It has nothing to do with `virtualenv` vs. `venv`. Old setuptools versions seem to ignore the `console_scripts` entry point and do not generate `venv/bin/pytest`. I'll drop this change.

Comment (Bastian-Krause) [reply .github/workflows/tests.yml]: Done.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (Bastian-Krause) [reply .github/workflows/tests.yml:44]: I've added that bit of info to the commit message.

**Review by ejoerns — APPROVED** [review]
Looks fine for me now :+1:

**Pull request merged**